programs/quarter_board/communication.c

```diff
@@ -0,0 +1,163 @@
+#include "ch.h"
+#include "hal.h"
+
+#include <icu_lld.h>
+#include "chprintf.h"
+#include <inttypes.h>
+
+#include "communication.h"
+
+thread_t *communicationp;
+
+MESSAGE message;
+MESSAGE *msgp = &message;
+
+uint8_t value_to_char (uint8_t character) {
+    if(character > 9) {
+       character = character - 10 + 'A';
+    } else {
+       character = character + '0';
+    }
+    return character;
+}
+
+
+uint8_t char_to_value(uint8_t character) {
+    if(character >= 'A') {
+        character = character + 10 - 'A';
+    } else {
+        character = character - '0';
+    }
+    return character;
+}
+
+THD_WORKING_AREA(waSerialThread, 128);
+THD_FUNCTION(SerialThread, msgThread) {
+    static uint16_t command;
+    static uint8_t my_address;
+    static uint8_t quantity_of_slaves;

```

Comment:
This can be quite easily named as: `num_slaves` or `slaves_count`, however I'm okay with this naming as long as it will be consistent.